src/main/java/uk/gov/hmcts/reform/sandl.snlevents/repository/db/HearingPartRepository.java

```diff
@@ -13,4 +15,41 @@
 public interface HearingPartRepository extends JpaRepository<HearingPart, UUID> {
 
     List<HearingPart> findBySessionIn(Collection<Session> sessions);
+
+    @Query(nativeQuery = true,
+        value = "select cast(title as varchar(100)) as title, cast(hearings as int) as hearings, \n"
+            + "cast(minutes as int) as minutes \n"
+            + "from (\n"
+            + "select 1 as sorted, 'Require listing in the next 4 weeks' as title, \n"
+            + "count(hearing_part) as hearings, \n"
+            + "coalesce(ceiling(sum(duration)/60), 0) as minutes\n"
+            + "from hearing_part\n"
+            + "where session_id is null \n"
+            + "and schedule_end < DATE_TRUNC('day', now() + interval '4 week' + interval '1 day')\n"
+            + "union all\n"
+            + "select 2 as sorted, 'Require listing in the next 3 months' as title, \n"
+            + "count(hearing_part) as hearings, \n"
+            + "coalesce(ceiling(sum(duration)/60), 0) as minutes\n"
+            + "from hearing_part\n"
+            + "where session_id is null \n"
+            + "and schedule_end >= DATE_TRUNC('day', now() + interval '4 week' + interval '1 day')\n"
+            + "and schedule_end < DATE_TRUNC('day', now() + interval '3 month' + interval '1 day')\n"
+            + "union all\n"
+            + "select 3 as sorted, 'Require listing beyond the next 3 months' as title, \n"
+            + "count(hearing_part) as hearings, \n"
+            + "coalesce(ceiling(sum(duration)/60), 0) as minutes\n"
+            + "from hearing_part\n"
+            + "where session_id is null \n"
+            + "and schedule_end >= DATE_TRUNC('day', now() + interval '3 month' + interval '1 day')\n"
+            + "union all\n"
+            + "select 4 as sorted, 'Target schedule not provided' as title, \n"
+            + "count(hearing_part) as hearings, \n"
+            + "coalesce(ceiling(sum(duration)/60), 0) as minutes\n"
+            + "from hearing_part\n"
+            + "where session_id is null \n"
+            + "and (schedule_end is null)\n"
+            + ") a order by sorted"
```

Comment:
31 reasons why I hate SQL strings in Java code :(
but I know you have checked this so 👍

Reply:
👘